pkg/apis/core/v1/storagecluster.go

```diff
@@ -176,6 +178,13 @@ type NodeSpec struct {
 	CommonConfig
 }
 
+// CSISpec is used to define the CSI configurations
+type CSISpec struct {
+	Enabled                   bool  `json:"enabled,omitempty"`
+	InstallSnapshotController *bool `json:"installSnapshotController,omitempty"`
+	InstallSnapshotCRDs       *bool `json:"installSnapshotCRDs,omitempty"`
```

Comment:
Would it happen that users have the CRDs but not the controller? Should we just have one flag to enable snapshot controller? We can try to create the CRDs and ignore if they are already present.

Reply:
Sure